File: java/TapstreamTest/src/com/tapstream/sdk/TestRunner.java
package com.tapstream.sdk;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.MappedByteBuffer;
import java.nio.channels.FileChannel;
import java.nio.charset.Charset;
import java.util.HashSet;
import java.util.Locale;
import java.util.Set;

import junit.framework.TestCase;

import org.mozilla.javascript.Context;
import org.mozilla.javascript.Scriptable;
import org.mozilla.javascript.ScriptableObject;

public class TestRunner {
	public class Util {
		private Context context;
		private ScriptableObject scope;

		public Util(Context context, ScriptableObject scope) {
			this.context = context;
			this.scope = scope;
		}

		public void fail(String message) {
			TestCase.fail(message);
		}

		public void assertEqual(Object a, Object b) {
			TestCase.assertEquals(a, b);
		}

		public void assertTrue(boolean b) {
			TestCase.assertTrue(b);
		}

		public void log(String message) {
			System.out.println(message);
		}

		public String getPostData(Tapstream ts) {
			return ts.core.getPostData();
		}

		public double getDelay(Tapstream ts) {
			return ts.core.getDelay();
		}

		public Scriptable getSavedFiredList(Tapstream ts) {
			Set<String> set = ((PlatformImpl) ts.platform).savedFiredList;
			if (set == null) {
				set = new HashSet<String>();
			}
			return context.newArray(scope, set.toArray());
		}

		public void setResponseStatus(Tapstream ts, int status) {
			((PlatformImpl) ts.platform).response = new Response(status, String.format(Locale.US, "Http %d", status));
		}

		public OperationQueue newOperationQueue() {
			return new OperationQueue();
		}

		public Config newConfig() {
			return new Config();
		}

		public Tapstream newTapstream(OperationQueue queue, String accountName, String secret, Config config) {
			return new Tapstream(queue, accountName, secret, config);
		}

		public Event newEvent(String name, boolean oneTimeOnly) {
			return new Event(name, oneTimeOnly);
		}

		public Hit newHit(String trackerName) {
			return new Hit(trackerName);
		}
	};

	public void run(String[] args) {
		String script = null;
		try {
			FileInputStream stream = new FileInputStream(new File(args[0]));
			try {
				FileChannel fc = stream.getChannel();
				MappedByteBuffer bb = fc.map(FileChannel.MapMode.READ_ONLY, 0, fc.size());
				script = Charset.availableCharsets().get("UTF-8").decode(bb).toString();
			} finally {
				stream.close();
			}
		} catch (Exception ex) {
			ex.printStackTrace();
			System.exit(1);
		}
		Context cx = Context.enter();
		try {
			ScriptableObject scope = cx.initStandardObjects();
			scope.putConst("language", scope, "java");
			scope.put("util", scope, new Util(cx, scope));
			cx.evaluateString(scope, script, args[0], 1, null);
		} catch (Error ex) {
			ex.printStackTrace();
			System.exit(1);
		} catch (Exception ex) {
			ex.printStackTrace();
			System.exit(1);
		} finally {
			Context.exit();
		}
		System.exit(0);
	}

	public static void main(String[] args) throws IOException {
		TestRunner tr = new TestRunner();
		tr.run(args);
	}
};

File: cs/TapstreamTest/TestRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Jint;
using Jint.Native;

namespace TapstreamMetrics.Sdk
{
    public class Util
    {
        private JintEngine engine;

        public Util(JintEngine engine)
        {
            this.engine = engine;
        }

        public void fail(string message)
        {
            Assert.Fail(message);
        }

        public void assertEqual(object a, object b)
        {
            Assert.AreEqual(a, b);
        }

        public void assertTrue(bool b)
        {
            Assert.IsTrue(b);
        }

        public void log(string message)
        {
            Console.Out.WriteLine(message);
        }

        public void setResponseStatus(Tapstream ts, int status)
        {
            ts.SetResponseStatus(status);
        }

        public string getPostData(Tapstream ts)
        {
            return ts.GetPostData();
        }

        public void setDelay(Tapstream ts, double delay)
        {
            ts.SetDelay((int)delay);
        }

        public double getDelay(Tapstream ts)
        {
            return ts.GetDelay();
        }

        public JsArray getSavedFiredList(Tapstream ts)
        {
            string[] strings = ts.GetSavedFiredList();
            JsArray array = engine.Global.ArrayClass.New();
            array.Length = strings.Length;
            for(int i = 0; i < strings.Length; i++)
            {
                array.put(i, engine.Global.StringClass.New(strings[i]));
            }
            return array;
        }

        public OperationQueue newOperationQueue()
        {
            return new OperationQueue();
        }

        public Config newConfig()
        {
            return new Config();
        }

        public Tapstream newTapstream(OperationQueue queue, string accountName, string secret, Config config)
        {
            return new Tapstream(queue, accountName, secret, config);
        }

        public Event newEvent(string name, bool oneTimeOnly)
        {
            return new Event(name, oneTimeOnly);
        }

        public Hit newHit(string trackerName)
        {
            return new Hit(trackerName);
        }
    }

    class ConsoleLogger : Logger
    {
        public void Log(LogLevel level, string msg)
        {
            Console.Out.WriteLine(msg);
        }
    }

    public class TestRunner
    {
        public static void Main(String[] args)
        {
            Logger logger = new ConsoleLogger();
            Logging.SetLogger(logger);

            string script = System.IO.File.ReadAllText(args[0]);
            JintEngine engine = new JintEngine();
            engine.SetParameter("language", "cs");
#if TEST_WINPHONE
            engine.SetParameter("platform", "winphone");
#else
            engine.SetParameter("platform", "windows");
#endif
            engine.SetParameter("util", new Util(engine));
            engine.Run(script);
        }
    }
}
